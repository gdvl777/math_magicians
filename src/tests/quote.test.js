import React from 'react';
import { render, waitFor } from '@testing-library/react';
import Quote from '../components/Quote';

// Mock the fetch function to simulate a successful response
beforeAll(() => {
  global.fetch = jest.fn(() =>
    Promise.resolve({ ok: true, json: () => Promise.resolve([{ quote: 'Test Quote' }]), }) )
});

afterAll(() => {
  global.fetch.mockClear();
  delete global.fetch;
});

test('renders the Quote component', async () => {
  // Render the Quote component
  const { getByText } = render(<Quote />);

  // Wait for the component to fetch data and render
  await waitFor(() => {
    expect(global.fetch).toHaveBeenCalled();
  });

  // Check if the loading text is not displayed
  expect(getByText('Loading...')).not.toBeInTheDocument();

  // Check if the quote is displayed
  const quoteElement = getByText('Test Quote');
  expect(quoteElement).toBeInTheDocument();
});
