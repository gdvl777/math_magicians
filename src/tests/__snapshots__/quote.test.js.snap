// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote renders the default quote 1`] = `
<div>
  <div
    class="loading"
  >
    Loading...
  </div>
</div>
`;
